web/impact/impact/tests/test_create_edit_office_hours_view.py

```diff
@@ -0,0 +1,244 @@
+                                                   minutes_from_now=-120)
+        self._assert_update_office_hour(mentor, office_hour, staff=True)
+
+    def test_mail_to_mentor_when_staff_create_office_hour_session(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        self._create_office_hour_session(self.staff_user(), data)
+        self.assertEqual(mail.outbox[0].to, [mentor.email])
+
+    def test_mail_to_mentor_when_staff_updates_office_hour_session(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        office_hour = self._create_office_hour_obj(mentor)
+        data = {'topics': self.updated_topics}
+        self._edit_office_hour_session(self.staff_user(), office_hour, data)
+        self.assertEqual(mail.outbox[0].to, [mentor.email])
+
+    def test_mentor_not_in_active_program_cannot_create_office_hour(self):
+        mentor = self._expert_with_inactive_program(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        with self._assert_office_hour_created(created=False):
+            self._create_office_hour_session(mentor, data)
+
+    def test_mentor_in_non_active_program_cant_create_office_response(self):
+        mentor = self._expert_with_inactive_program(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        response = self._create_office_hour_session(mentor, data)
+        self.assertEqual(response.data['detail'],
+                         DEFAULT_PERMISSION_DENIED_DETAIL)
+
+    def test_office_hour_start_date_must_be_later_than_start_date(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        start_time = datetime.now()
+        data = self._get_post_request_data(mentor, get_data={
+            'start_date_time': start_time,
+            'end_date_time': start_time + timedelta(minutes=-30),
+        })
+        response = self._create_office_hour_session(mentor, data)
+        self._assert_error_response(response.data,
+                                    key='end_date_time',
+                                    expected=INVALID_END_DATE)
+
+    def test_none_staff_or_none_mentor_response(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor)
+        response = self._create_office_hour_session(self.basic_user(), data)
+        self.assertEqual(response.data['detail'],
+                         DEFAULT_PERMISSION_DENIED_DETAIL)
+
+    def test_use_request_user_for_mentor_users(self):
+        mentor = self._expert_user(UserRole.MENTOR)
+        mentor2 = self._expert_user(UserRole.MENTOR)
+        data = self._get_post_request_data(mentor,
+                                           get_data={'mentor': mentor2.id})
+        response = self._create_office_hour_session(mentor, data)
+        self.assertEqual(response.data['data']['mentor']['id'], mentor.id)
+
+    def test_admin_cant_create_office_hour_for_restricted_user_response(self):
```

Comment:
What is a "restricted user"? If this is just "judge" or "non-mentor", let's use the more precise language.